### PR TITLE
[zh-CN] add translation for Object-oriented_programming

### PR DESCRIPTION
**Pull request by oolongcha024**
### Description

- add translation for the Object-oriented_programming
- fixed incorrect link on prev-next menu in the object_prototypes

### CODE REVIEW

File: files/zh-cn/learn/javascript/objects/object-oriented_programming/index.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+title: 面向对象编程基本概念
+slug: Learn/JavaScript/Objects/Object-oriented_programming
+tags:
+  - JavaScript
+  - Learn
+---
+
+{{LearnSidebar}}{{PreviousMenuNext("Learn/JavaScript/Objects/Object_prototypes", "Learn/JavaScript/Objects/Classes_in_JavaScript", "Learn/JavaScript/Objects")}}
+
+面向对象编程（OOP）是如今多种编程语言所实现的一种编程范式，包括 Java、C++。在本文中，我们会简要介绍面向对象编程（OOP）的基本概念，其中包括三个主要概念：**类与实例**、**继承**、**封装**。现在，我们会脱离特定的 JavaScript 代码来探讨这些概念，所有提及到的例子将会 {{Glossary("Pseudocode", "伪代码")}} 的形式描述。
```
Comment (fwqaaq) [inline files/zh-cn/learn/javascript/objects/object-oriented_programming/index.md:11]:
```suggestion
面向对象编程（OOP）是如今多种编程语言所实现的一种编程范式，包括 Java、C++。在本文中，我们会简要介绍面向对象编程（OOP）的基本概念，其中包括三个主要概念：**类与实例**、**继承**、**封装**。现在，我们会脱离特定的 JavaScript 代码来探讨这些概念，所有提及到的例子将会以{{Glossary("Pseudocode", "伪代码")}}的形式描述。
```

File: files/zh-cn/learn/javascript/objects/object-oriented_programming/index.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
+---
+title: 面向对象编程基本概念
+slug: Learn/JavaScript/Objects/Object-oriented_programming
+tags:
+  - JavaScript
+  - Learn
```
Comment (yin1999) [inline files/zh-cn/learn/javascript/objects/object-oriented_programming/index.md:6]: Hi @tgz033. We just do not need this in translated document. See: https://github.com/mdn/translated-content/blob/main/docs/zh-cn/translation-guide.md#%E5%85%83%E6%95%B0%E6%8D%AE

File: files/zh-cn/learn/javascript/objects/object-oriented_programming/index.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+---
+
+{{LearnSidebar}}{{PreviousMenuNext("Learn/JavaScript/Objects/Object_prototypes", "Learn/JavaScript/Objects/Classes_in_JavaScript", "Learn/JavaScript/Objects")}}
+
+面向对象编程（OOP）是如今多种编程语言所实现的一种编程范式，包括 Java、C++。在本文中，我们会简要介绍面向对象编程（OOP）的基本概念，其中包括三个主要概念：**类与实例**、**继承**、**封装**。现在，我们会脱离特定的 JavaScript 代码来探讨这些概念，所有提及到的例子将会以{{Glossary("Pseudocode", "伪代码")}}的形式描述。
+
+> **备注：** 准确地说，本文所提及到的特性是一种特别的面向对象编程方式，即**基于类的面向对象编程**（class-based OPP）。当人们谈论面向对象编程时，通常来说是指基于类的面向对象编程。
+
+在本文的最后，我们会探讨 JavaScript 中的构造器和原型链是如何与面向对象编程中的概念产生联系的，以及它们与面向对象编程中的概念又有何不同。在下一篇文章中，我们会学习 JavaScript 中一些附加的特性，这些特性使得实现面向对象编程变得更加容易。
```
Comment (yin1999) [inline files/zh-cn/learn/javascript/objects/object-oriented_programming/index.md:15]: We prefer `构造函数`

File: files/zh-cn/learn/javascript/objects/object-oriented_programming/index.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    <tr>
+      <th scope="row">目标：</th>
+      <td>
+        理解基于类的面向对象编程中的基础概念。
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+面向对象编程将一个系统抽象为许多对象的集合，每一个对象代表了这个系统的特定方面。对象包括函数（方法）和数据。一个对象可以向其他部分的代码提供一个公共接口，而其他部分的代码可以通过公共接口执行该对象的特定操作，系统的其他部分不需要关心对象内部是如何完成任务的，这样保持了对象自己内部状态的私有性（译者注：也就是说，外部代码无法随意修改对象的内部状态）。
```
Comment (yin1999) [inline files/zh-cn/learn/javascript/objects/object-oriented_programming/index.md:42]: If not necessary, the translator's note could be removed.

**Review by yin1999 — APPROVED** [review]
Thanks @tgz033. I've made some fixes on this PR. Thanks for your contribution. Welcome aboard 🎉

**Pull request merged**